### ISSUE TITLE
ZAP on Jenkins not showing high priority vulnerabilities

### ISSUE THREAD

**Issue by akashlomas** [zaproxy/zaproxy#4733]
Hi,

We are using ZAP on Jenkins to execute security vulnerability scans on websites. We are getting results which are not expected results.

When we run ZAP 2.7.0 on http://testasp.netvulnweb.com, I get 3 major, 1 medium and 2 low priority vulnerabilities.

![zap-m-1](https://user-images.githubusercontent.com/633607/41047909-397f6fe0-69cb-11e8-93d2-caebb7a3c09e.PNG)
 
But, when I run the same site against ZAP on Jenkins, I get only 1 medium and 2 low priority vulnerabilities.
 
![zap-0](https://user-images.githubusercontent.com/633607/41048088-a23fffa4-69cb-11e8-88fe-511b97a4e2fd.PNG)

I have setup following ZAP plugins installed on my Jenkins;

![zap-4](https://user-images.githubusercontent.com/633607/41048171-c8e7b520-69cb-11e8-98e1-cd48ebd17b30.PNG)

and the Jenkins task is configured as;

![zap-1](https://user-images.githubusercontent.com/633607/41048114-ab341d5c-69cb-11e8-984b-323549c6ea73.PNG)

![zap-2](https://user-images.githubusercontent.com/633607/41048125-af78750c-69cb-11e8-8822-9cbaa523c6d9.PNG)

![zap-3](https://user-images.githubusercontent.com/633607/41048150-b6668dea-69cb-11e8-9460-e42b7b411209.PNG)

The result log is also attached.
[ZAP-Jenkins-Results.txt](https://github.com/zaproxy/zaproxy/files/2076929/ZAP-Jenkins-Results.txt)

Kindly share your feedback as we would like to run ZAP as vulnerability scanner for OWASP top 10 vulnerabilities.

Thanks

**Comment by thc202** [comment]
The context is just including one URL, it should include more to allow to spider/scan other pages (e.g. `Comments.aspx`), something like: `http://testaspnet.vulnweb.com/.*`

Note that you can uninstall the other ZAP plugin (it was superseeded by the official plugin).

**Comment by thc202** [comment]
@akashlomas is this resolved?

**Comment by akashlomas** [comment]
no @thc202, no luck so far  :(

**Comment by akashlomas** [comment]
I have tried ZAP tool acting as Proxy while I go through every page of website and provide all necessary information on pages. Once this is done, I run spider followed by active scan. I get good results with this type of scan.

But with ZAP plugin in my CI-CD pipeline, I don't  get same set of results.

Specially SQL injection is only caught while I test with first method.

**Comment by thc202** [comment]
Are you using the same ZAP add-ons (and same scan policy) in Jenkins?

**Comment by akashlomas** [comment]
yes I am, please let me know if I can share any file/ config or screenshot (except already posted above) which will help to resolve this.

**Comment by kingthorin** [comment]
The thing is since you manually prime things when using the GUI. That kind of impacts the comparison. For the comparison to be equivalent you need to do the exact same actions with the exact same config/values.

**Comment by kingthorin** [comment]
@akashlomas any update?

**Comment by kingthorin** [comment]
1) Do you have the same set of addons installed?
2) Are you testing the exact same set of URLs/requests?

**Comment by kooolbreez1** [comment]
Here is my Zap plugin configuration in case this helps:

![image](https://user-images.githubusercontent.com/7256563/42957203-f0485e9e-8b4f-11e8-8010-7c756864401d.png)

and my Jenkins console output:

![image](https://user-images.githubusercontent.com/7256563/42957291-334f3730-8b50-11e8-9c6e-2783fdbc0c00.png)

and my results from the scan (in Jenkins, for some reason I cant take a screenshot of the full report but the summary of alerts is different from the desktop version):

![image](https://user-images.githubusercontent.com/7256563/42957596-2aed9b30-8b51-11e8-9e31-9cc50f1cea85.png)

finally, my results from the desktop version of ZAP (running on macOS Mojave public beta 2):

<img width="394" alt="screen shot 2018-07-19 at 12 49 04 pm" src="https://user-images.githubusercontent.com/7256563/42957992-3d761f9c-8b52-11e8-9851-b4c95e962638.png">

I'll be honest, I'm a total noob when it comes to using both Zap and Jenkins so let me know if there is anything else I can provide that might be helpful. I don't really want to hijack this post but I feel like my issue is basically the same as @akashlomas so I posted here.

**Comment by thc202** [comment]
The include in context should end in `.*` to include URLs under that one. Could you try changing that?

**Comment by kooolbreez1** [comment]
Wow, I totally misread your post higher up and thought I just had to add `*` instead of `.*`, I feel like an idiot but that totally fixed it. Thanks a lot :)

**Comment by kingthorin** [comment]
Thanks for letting us know.
